### PR TITLE
Removed logger kwarg from JobResult.log()

### PR DESCRIPTION
**Pull request by gsnider2195**
# Closes: #3892 
# What's Changed

- Removed `logger` kwarg from `JobResult.log()`. Jobs' `self.logger` logging instance automatically logs to both the console and database and should be used instead of this method.
- Changed the function signatures for `nautobot.extras.datasources.git` methods `get_repo_from_url_to_path_and_from_branch`, `ensure_git_repository`, `git_repository_dry_run`, `import_config_context`, `import_local_config_context`, and `import_config_context_schema`.
- Removed unused method `get_job_result_and_repository_record` in `nautobot.extras.datasources.git`.

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/extras/datasources/git.py [hunk 1]
```diff
@@ -271,34 +216,29 @@ def update_git_config_contexts(repository_record, job_result):
...
         if not os.path.isfile(os.path.join(config_context_path, file_name)):
             continue
-        job_result.log(
-            f"Loading config context from `{file_name}`",
-            grouping="config contexts",
-            logger=logger,
-        )
+        msg = f"Loading config context from `{file_name}`"
+        logger.info(msg)
+        job_result.log(msg, grouping="config contexts")
```
Comment (HanlinMiao) [inline nautobot/extras/datasources/git.py:221]: Do we need the double logging here? The same question throughout.

Comment (gsnider2195) [reply nautobot/extras/datasources/git.py:221]: The first line logs to the console, the second line logs to the database as a `JobLogEntry`

File: changes/3892.changed [hunk 2]
```diff
@@ -0,0 +1 @@
+Removed `logger` kwarg from `JobResult.log()`.
```
Comment (gsnider2195) [inline changes/3892.changed:1]:
```suggestion
Removed `logger` kwarg from `JobResult.log()`. Jobs' `self.logger` logging instance now performs this behavior automatically.
```

File: nautobot/extras/datasources/utils.py [hunk 3]
```diff
@@ -29,12 +29,9 @@ def files_from_contenttype_directories(base_path, job_result, log_grouping):
...
             except ContentType.DoesNotExist:
-                job_result.log(
-                    f"Skipping `{app_label}.{modelname}` as it isn't a known content type",
-                    level_choice=LogLevelChoices.LOG_WARNING,
-                    grouping=log_grouping,
-                    logger=logger,
-                )
+                msg = f"Skipping `{app_label}.{modelname}` as it isn't a known content type"
+                logger.warning(msg)
+                job_result.log(msg, level_choice=LogLevelChoices.LOG_WARNING, grouping=log_grouping)
```
Comment (bryanculver) [inline nautobot/extras/datasources/utils.py:34]: Is this redundant?

Comment (gsnider2195) [reply nautobot/extras/datasources/utils.py:34]: The first line logs to the console, the second line logs to the database as a `JobLogEntry`

**Comment by gsnider2195** [comment]
Should this change be documented in the app developer migration guide that @HanlinMiao is working on in #4156?

File: nautobot/extras/datasources/git.py [hunk 4]
```diff
@@ -738,8 +624,8 @@ def update_git_config_context_schemas(repository_record, job_result):
 
 
 def import_config_context_schema(
-    context_schema_data, repository_record, job_result, logger  # pylint: disable=redefined-outer-name
-):
+    context_schema_data, repository_record, job_result
+):  # pylint: disable=redefined-outer-name
```
Comment (glennmatthews) [inline nautobot/extras/datasources/git.py:628]: is this disable still needed after removing the `logger` param?

Comment (glennmatthews) [reply nautobot/extras/datasources/git.py:628]: I don't fully understand why `import_local_config_context()` still has a `logger` param, while `import_config_context()` and `import_config_context_schema()` do not and instead now use the global `logger`? What's the pattern here that I'm missing?

Comment (gsnider2195) [reply nautobot/extras/datasources/git.py:628]: That was an oversight on my part. The only logger args that need to stay are for the `ensure_git_repository` and `git_repository_dry_run` methods that are both called by jobs.

**Review by glennmatthews — APPROVED** [review]

**Pull request merged**